client/dashboard/emails/add-professional-email/index.tsx

```diff
@@ -0,0 +1,259 @@
+		// @ts-expect-error the query is only enabled when domain has a value, so blog_id won't be undefined
+		...mailboxAccountsQuery( domain?.blog_id, domainName ),
+		enabled: !! domain,
+	} );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ mailboxEntities, setMailboxEntities ] = useState<
+		MailboxFormEntity< SupportedEmailProvider >[]
+	>( [] );
+
+	useEffect( () => {
+		if ( isDomainFetched && ! domain ) {
+			router.navigate( { to: '/emails' } );
+		}
+	}, [ domain, isDomainFetched, router ] );
+
+	const isDomainInCart = false; // TODO: This can be set as a prop if we implement `EmailProvidersUpsell`
+
+	const createNewMailbox = useCallback( () => {
+		const mailbox = new MailboxFormEntity< SupportedEmailProvider >(
+			'titan',
+			domainName,
+			( existingMailboxes ?? [] )
+				.flatMap( ( emailAccount ) => emailAccount.emails )
+				.map( ( emailBox ) => emailBox.mailbox )
+		);
+
+		possibleHiddenFieldNames.forEach( ( fieldName ) => {
+			mailbox.setFieldIsVisible( fieldName, false );
+			mailbox.setFieldIsRequired( fieldName, false );
+		} );
+
+		// Set initial values
+		Object.entries( {
+			[ FIELD_PASSWORD_RESET_EMAIL ]: user.email,
+		} ).forEach( ( [ fieldName, value ] ) => {
+			mailbox.setFieldValue( fieldName as FormFieldNames, value );
+		} );
+
+		return mailbox;
+	}, [ domainName, existingMailboxes, user.email ] );
+
+	useEffect( () => {
+		isFetched && setMailboxEntities( [ createNewMailbox() ] );
+	}, [ createNewMailbox, isFetched ] );
+
+	const handleSubmit = async ( e: FormEvent< HTMLFormElement > ) => {
+		e.preventDefault();
+
+		const { shoppingCartManagerClient } = await import(
+			/* webpackChunkName: "async-load-shopping-cart" */ '../../app/shopping-cart'
+		);
+
+		mailboxEntities.forEach( ( mailbox ) => mailbox.validate( true ) );
+		const mailboxOperations = new MailboxOperations( mailboxEntities, () => {} );
+
+		setIsSubmitting( true );
+
+		const validated = await mailboxOperations.validateAndCheck( false );
+
+		if ( ! userCanAddEmail || ! validated ) {
+			if ( ! userCanAddEmail ) {
+				const errors = domain?.current_user_cannot_add_email_reason?.errors;
+				const message = errors
+					? sprintf(
+							// Translators: %(errors)s is a list of errors separated by commas.
+							__( 'You cannot add emails to this domain: %(errors)s.' ),
+							{ errors: Object.values( errors ).join( ', ' ) }
+					  )
+					: __( 'You cannot add emails to this domain.' );
+				createErrorNotice( message, { type: 'snackbar' } );
+			}
+
+			setIsSubmitting( false );
+
+			return;
+		}
+
+		const numberOfMailboxes = mailboxOperations.mailboxes.length;
+
+		const emailProperties = getEmailProductProperties(
+			'titan',
+			domain,
+			product,
+			numberOfMailboxes
+		);
+
+		const checkoutPath = getEmailCheckoutPath(
+			domainName,
+			domain.domain,
+			router.state.location.pathname,
+			mailboxOperations.mailboxes[ 0 ].getAsCartItem().email
+		);
```

Comment:
This why @rcrdortiz  sees the site selector [here](https://github.com/Automattic/wp-calypso/pull/106524#issuecomment-3415416650).

It can be reproduced on secondary domains:

- Purchase two domains for a given site
- Pick one of them and make it the primary domain
- If you go to `/v2/emails/add-professional-email/:primaryDomain`, you can proceed to checkout normally
- If you go to `/v2/emails/add-professional-email/:secondaryDomain`, the checkout fails and you are redirected to the site selector

I think it can be fixed by using the site slug in the checkout URL:

```suggestion
		const checkoutPath = getEmailCheckoutPath(
			site.slug,
			domain.domain,
			router.state.location.pathname,
			mailboxOperations.mailboxes[ 0 ].getAsCartItem().email
		);
```

Reply:
@mmtr @rcrdortiz you'r right! Fixed on https://github.com/Automattic/wp-calypso/pull/106524/commits/5b9eb7b6975278c510879c1896d540e0d00d2848